File: packages/ilc-py/src/devalbo_ilc/examples/hello.py
"""Run: python -m devalbo_ilc.examples.hello"""

from __future__ import annotations

import asyncio
import sys

from devalbo_ilc.hello import hello
from devalbo_ilc.hosts import create_process_environment, create_test_environment


async def run() -> None:
    if "--test" in sys.argv:
        env = create_test_environment()
        await hello(env)
        info = next((e for e in env.logs if e.level == "info"), None)
        if info is None or info.message != "hello from ILC":
            print("unexpected logs:", env.logs, file=sys.stderr)
            raise SystemExit(1)
        print("ok: test host captured hello", file=sys.stderr)
        return

    await hello(create_process_environment())


def main() -> None:
    asyncio.run(run())


if __name__ == "__main__":
    main()

File: packages/ilc-rs/examples/hello.rs
use devalbo_ilc::{create_process_environment, create_test_environment, hello};

#[tokio::main]
async fn main() {
    let args: Vec<String> = std::env::args().collect();
    if args.iter().any(|a| a == "--test") {
        let env = create_test_environment(vec![]);
        hello(&env).await;
        let logs = env.logs.lock().unwrap();
        let ok = logs
            .iter()
            .any(|e| e.level == "info" && e.message == "hello from ILC");
        if !ok {
            eprintln!("unexpected logs: {logs:?}");
            std::process::exit(1);
        }
        eprintln!("ok: test host captured hello");
        return;
    }

    let env = create_process_environment();
    hello(&env).await;
}
